src/woo_publications/api/pagination.py

```diff
@@ -0,0 +1,10 @@
+from django.conf import settings
+
+from rest_framework.pagination import PageNumberPagination
+
+max_page_size = settings.REST_FRAMEWORK["MAX_PAGE_SIZE"]
+
+
+class DynamicPageSizePagination(PageNumberPagination):
+    page_size_query_param = "page_size"
+    max_page_size = max_page_size
```

Comment:
```suggestion

class DynamicPageSizePagination(PageNumberPagination):
    page_size_query_param = "page_size"
    max_page_size = 100
```
See the remark below about inventing custom DRF settings.